misc/toml-parser/src/main/java/io/ballerina/toml/validator/schema/SchemaDeserializer.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.toml.validator.schema;
+
+import com.google.gson.JsonDeserializationContext;
+import com.google.gson.JsonDeserializer;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * Responsible for Deserializing abstract Schema nodes accordingly.
+ *
+ * @since 2.0.0
+ */
+public class SchemaDeserializer implements JsonDeserializer<AbstractSchema> {
+
+    @Override
+    public AbstractSchema deserialize(JsonElement jsonElement, java.lang.reflect.Type refType,
+                                      JsonDeserializationContext jsonDeserializationContext)
+            throws JsonParseException {
+        JsonObject jsonObj = jsonElement.getAsJsonObject();
+        String type = jsonObj.get("type").getAsString();
+        switch (type) {
+            case "object":
+                return getObjectSchema(jsonDeserializationContext, jsonObj);
+            case "array":
+                return getArraySchema(jsonDeserializationContext, jsonObj);
+            case "integer":
+                return getNumericSchema(jsonObj, Type.INTEGER);
+            case "number":
+                return getNumericSchema(jsonObj, Type.NUMBER);
+            case "string":
+                return getStringSchema(jsonObj);
+            case "boolean":
+                return new BooleanSchema(Type.BOOLEAN);
+            default:
+                throw new JsonParseException("type " + type + " is not supported type");
```

Comment:
We should throw our own exception here. Throwing an exception from  the `gson` library is not correct.